Description: Fix CMake compatibility for CMake 4.x
 CMake 4.x removed compatibility with cmake_minimum_required < 3.5
 and removed support for CMP0042 OLD policy.
Author: Cuihtlauac ALVARADO <cuihtlauac.alvarado@gmail.com>
Last-Update: 2026-02-02

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)

 # setup directory where we should look for cmake files
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
@@ -16,10 +16,11 @@ project(libjson-rpc-cpp)
 cmake_policy(SET CMP0007 NEW)
 cmake_policy(SET CMP0012 NEW)

-if (${CMAKE_MAJOR_VERSION} GREATER 2)
-    # old policy do not use MACOSX_RPATH
-    cmake_policy(SET CMP0042 OLD)
-endif()
+# Use MACOSX_RPATH by default (modern CMake behavior)
+if(POLICY CMP0042)
+    cmake_policy(SET CMP0042 NEW)
+endif()
+set(CMAKE_MACOSX_RPATH ON)

 set(MAJOR_VERSION 1)
 set(MINOR_VERSION 4)
